renku/core/utils/requests.py

```diff
@@ -33,4 +36,7 @@ def retry(total_requests=10, backoff_factor=1, statuses=(500, 502, 503, 504, 429
     _session.mount("http://", HTTPAdapter(max_retries=retries))
     _session.mount("https://", HTTPAdapter(max_retries=retries))
 
-    yield _session
+    try:
+        yield _session
+    except (ConnectionError, RequestException) as e:
```

Comment:
`RequestException` is the base exception class and includes exceptions which are related to connection failure (e.g. `HTTPError`). Maybe here we should catch `ConnectionError` and `Timeout`: https://requests.readthedocs.io/en/master/_modules/requests/exceptions/

Reply:
Yea, this evolved without too many reviews on my side. I think this should capture only `RequestException` actually because the problem is that we are not re-raising network issues as renku exceptions. It is much simpler to deal with the exception on the client-side if the client knows that any exception which will bubble up from the renku core module will be of type `RenkuException`, hence the reraise here fixes the problem (I believe).

Reply:
> I think this should capture only `RequestException` actually